ghost/core/core/server/services/settings-helpers/SettingsHelpers.js

```diff
@@ -220,16 +222,46 @@ class SettingsHelpers {
     }
 
     /**
-     * Social web (ActivityPub) is enabled if:
-     * - Social web is enabled in the settings
-     * - 'ActivityPub' flag is enabled in the labs, for compatibility with Ghost 5.x
-     * - Config allows it (TODO)
-     * - Billing allows it (TODO)
+     * Calculated setting for Social web (ActivityPub)
      *
      * @returns {boolean}
      */
     isSocialWebEnabled() {
-        return this.settingsCache.get('social_web') === true && this.labs.isSet('ActivityPub');
+        // UI setting
+        if (this.settingsCache.get('social_web') !== true) {
+            logging.warn('Social web is disabled in settings');
+            return false;
+        }
+
+        // Labs setting
+        if (!this.labs.isSet('ActivityPub')) {
+            logging.warn('Social web is disabled in labs');
+            return false;
+        }
+
+        // Ghost (Pro) limits
+        if (this.limitService.isDisabled('limitSocialWeb')) {
+            logging.warn('Social web is not available for Ghost (Pro) sites without a custom domain, or hosted on a subdirectory');
+            return false;
+        }
+
+        // Social web (ActivityPub) currently does not support Ghost sites hosted on a subdirectory, e.g. https://example.com/blog/
+        const subdirectory = this.urlUtils.getSubdir();
+        if (subdirectory) {
+            logging.warn('Social web is not available for Ghost sites hosted on a subdirectory');
+            return false;
+        }
+
+        // Self-hosters cannot connect to production ActivityPub servers from localhost or IPs addresses
+        const siteUrl = new URL(this.urlUtils.getSiteUrl());
+        const isLocalhost = siteUrl.hostname === 'localhost' || siteUrl.hostname === '127.0.0.1' || siteUrl.hostname === '::1';
+        const isIP = net.isIP(siteUrl.hostname);
+        if (process.env.NODE_ENV === 'production' && (isLocalhost || isIP)) {
+            logging.warn('Social web is not available for localhost or IPs addresses in production');
+            return false;
+        }
+
+        return true;
```

Comment:
I really like the structure of the logic here 👍 

However, I feel like having WARN level logging every time the setting is evaluated is overkill. Using debug would be great here instead and we should trust the UI to communicate with the user.

Reply:
Updated in 9b8297a4901b68c301625995e5fe7f7c7a6afffb